controller/src/main/java/org/jboss/as/controller/ThreadDumpUtil.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2011, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.controller;
+
+import org.jboss.as.controller.logging.ControllerLogger;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.management.LockInfo;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+
+final class ThreadDumpUtil {
+
+   public static void threadDump() {
+
+      try (
+         StringWriter dumpstr = new StringWriter();
+         PrintWriter dumpout = new PrintWriter(dumpstr);
+         StringWriter deadlockstr = new StringWriter();
+         PrintWriter deadlockout = new PrintWriter(deadlockstr);
+      ) {
+
+         ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
+
+         for (ThreadInfo threadInfo : threadMXBean.dumpAllThreads(true, true)) {
+            dumpout.println(threadInfoToString(threadInfo));
+         }
+
+         ControllerLogger.MGMT_OP_LOGGER.threadDump(dumpstr.toString());
+
+         long[] deadlockedThreads = threadMXBean.findDeadlockedThreads();
+
+         if (deadlockedThreads != null && deadlockedThreads.length > 0) {
+            for (ThreadInfo threadInfo : threadMXBean.getThreadInfo(deadlockedThreads, true, true)) {
+               deadlockout.println(threadInfoToString(threadInfo));
+            }
+
+            ControllerLogger.MGMT_OP_LOGGER.deadLock(deadlockstr.toString());
+         }
+
+      } catch (IOException e) {
```

Comment:
Would it be better to catch `Exception` instead so we can also capture any `RuntimeExcepion`?